Assets/Scripts/Animator/AvatarAnimator.cs

```diff
@@ -0,0 +1,132 @@
+// Copyright 2021-2022 The SeedV Lab.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using Mediapipe;
+using Mediapipe.Unity;
+using UnityEngine;
+
+namespace SeedUnityVRKit {
+  // <summary>An animator to visualize upper body and face.</summary>
+  public class AvatarAnimator : MonoBehaviour {
```

Comment:
My preference is not to introduce the behavior in a separated class without referencing in the scene in this project.

What I prefer is:

1. Create virtual methods in `UpperBodyAnimator` that should be diverged from its current logic. For example, `SetEye`, `SetMouth`, and the `_neckInitRotation`. It may not be an exhausive list, so please make sure you go through all the differences in the current `UpperBodyAnimator` and `AvatarAnimator`. Note that for `_neckInitRotation`, maybe you need to move the initialization to the constructor.
2. In Studio project, create AvatarAnimator class to inherit `UpperBodyAnimator` and override these methods.
3. Keep `FaceController` class in Studio project. We don't need that.

Reply:
I say ，so I can change the UpperBodyAnimator function and it param.  you tell me only can move function don't change it in UpperBodyAinmator， I first want to change SetEye， SetMouth to virual function and override it。

Reply:
Yeah, I was consider step 2 to refactor `UpperBodyAnimator`, like extracting methods. The goal was to clearly separate the logic that should be common to `UpperBodyAnimator`, and the one that should be customized. This part is still the same.

Using `virtual` function and `override` in `AvatarAnimator` is a after-thought after looking at the current change. The other alternative is to create `if`-`else` blocks like in your first commit. I think virtual function would work better since it separates Avatar logic to its own class.